src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/request/AddQuestionWithResponses.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.incidentreporting.dto.request
+
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.constraints.Size
+import java.time.LocalDateTime
+
+@Schema(description = "Payload to add question with responses to an incident report")
+data class AddQuestionWithResponses(
+  @Schema(description = "The question code", required = true)
+  @field:Size(max = 60)
+  val code: String,
+  @Schema(description = "The question", required = true)
+  val question: String,
+  @Schema(description = "The responses to this question", required = true)
+  val responses: List<AddQuestionResponse>,
+  @Schema(description = "Optional additional information", required = false, defaultValue = "null")
+  val additionalInformation: String? = null,
+)
+
+data class AddQuestionResponse(
+  @Schema(description = "The response", required = true)
+  val response: String,
+  @Schema(description = "Username of person who responded to the question", required = true)
+  @field:Size(min = 3, max = 120)
+  val recordedBy: String,
+  @Schema(description = "When the response was made", required = true, example = "2024-04-29T12:34:56.789012")
+  val recordedAt: LocalDateTime,
```

Comment:
Any reason to ask the clients to pass this?

I'm assuming this API is used to submit questions so in theory we can set this to "now" when we persist them no?

At the very least we should make this optional and default to now if you think it's still valuable to allow clients to override this.

Reply:
it's a good question. i sorta propagated the pattern from the original `CreateReportRequest.reportedBy` without thinking too much about it. which do you think is better…
1) remove recordedBy and recordedAt here and use `authenticationFacade.getUserOrSystemInContext()` and `LocalDateTime.now(clock)` – because those 2 are used for the parent report's modifiedBy and modifiedAt
2) as you suggest, make them optional but _also_ propagate them to the report's modifiedBy and modifiedAt

Reply:
i dunno which is better :/

Reply:
nomis screenshots imply that it might be possible for different responses to 1 question to have been made by different users…

Reply:
To recap our chat: Consistency aside (which I agree is important for an API), I think my preference would be to not have these fields and set them automatically to "current real user"/"now".

When/_if_ the need to override these good default ever arise we can always update the API to have optional fields which still default to these values (and this would be a non-breaking change for existing clients).